### PR TITLE
add a new base layer without a toolchain

### PR DESCRIPTION
**Pull request by nashif**
Add a new base layer without toolchains to allow for downstream
customization and usage with non zephyr-sdk toolchains.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,6 +45,13 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v3
 
+    - name: Generate local metadata for CI image
```
Comment (stephanosio) [inline .github/workflows/ci.yml:48]:
```suggestion
    - name: Generate local metadata for base image
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -89,11 +111,21 @@ jobs:
           type=ref,event=branch
           type=ref,event=tag
 
-    - name: Set up Docker Buildx
+    - name: Set up Docker Build
```
Comment (stephanosio) [inline .github/workflows/ci.yml:114]: `Buildx` is not a typo here. We are, in fact, setting up the "Buildx" builder instance here.

```suggestion
    - name: Set up Docker Buildx
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -89,11 +111,21 @@ jobs:
...
 
+    - name: Build base docker image
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        file: Dockerfile.base
+        platforms: ${{ matrix.variant.platform }}
+        push: true
+        tags: ${{ steps.meta_ci.outputs.tags }}
+        labels: ${{ steps.meta_ci.outputs.labels }}
```
Comment (stephanosio) [inline .github/workflows/ci.yml:127]:
```suggestion
        tags: ${{ steps.meta_ci_base.outputs.tags }}
        labels: ${{ steps.meta_ci_base.outputs.labels }}
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -103,6 +135,8 @@ jobs:
         push: true
         tags: ${{ steps.meta_ci.outputs.tags }}
         labels: ${{ steps.meta_ci.outputs.labels }}
+        build-args: |
+          BASE_IMAGE=localhost:5000/zephyrproject-rtos/ci-base:${{ steps.meta_ci.outputs.version }}
```
Comment (stephanosio) [inline .github/workflows/ci.yml:139]:
```suggestion
          BASE_IMAGE=localhost:5000/zephyrproject-rtos/ci-base:${{ steps.meta_ci_base.outputs.version }}
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -132,6 +166,13 @@ jobs:
         username: ${{ github.repository_owner }}
         password: ${{ secrets.GITHUB_TOKEN }}
 
+    - name: Push base docker image
+      if: ${{ github.event_name != 'pull_request' }}
+      uses: stephanosio/tag-push-action@v2.1.0
+      with:
+        src: localhost:5000/zephyrproject-rtos/ci-base:${{ steps.meta_ci.outputs.version }}
+        dst: ${{ steps.meta_ci_push.outputs.tags }}
```
Comment (stephanosio) [inline .github/workflows/ci.yml:174]:
```suggestion
        src: localhost:5000/zephyrproject-rtos/ci-base:${{ steps.meta_ci_base.outputs.version }}
        dst: ${{ steps.meta_ci_base_push.outputs.tags }}
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -198,27 +253,38 @@ jobs:
     - name: Create multi-architecture image
       run: |
         archs=(amd64 arm64)
+        ci_base_image="ghcr.io/zephyrproject-rtos/ci-base:${{ steps.meta_ci_push.outputs.version }}"
```
Comment (stephanosio) [inline .github/workflows/ci.yml:256]:
```suggestion
        ci_base_image="ghcr.io/zephyrproject-rtos/ci-base:${{ steps.meta_ci_base_push.outputs.version }}"
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -198,27 +253,38 @@ jobs:
...
 
+        docker manifest push ${ci_base_image}
         docker manifest push ${ci_image}
         docker manifest push ${developer_image}
 
+    - name: Push base docker image
+      uses: stephanosio/tag-push-action@v2.1.0
+      with:
+        src: ghcr.io/zephyrproject-rtos/ci-base:${{ steps.meta_ci_push.outputs.version }}
+        dst: ${{ steps.meta_ci_push.outputs.tags }}
```
Comment (stephanosio) [inline .github/workflows/ci.yml:286]:
```suggestion
        src: ghcr.io/zephyrproject-rtos/ci-base:${{ steps.meta_ci_base_push.outputs.version }}
        dst: ${{ steps.meta_ci_base_push.outputs.tags }}
```

**Review by stephanosio — CHANGES_REQUESTED** [review]

**Comment by stephanosio** [comment]
Also should we rename `Dockerfile` to `Dockerfile.ci` now that we have three image types and it is just confusing without proper notation?

**Comment by stephanosio** [comment]
Also need to update the README as well.

**Review by galak — APPROVED** [review]

**Pull request merged**